### PR TITLE
1.28 Backport H/2 header discard fix

### PR DESCRIPTION
**Pull request by SeanKilleen**
Commit Message:
Discard the Host header if the :authority header was received to bring Envoy into compliance with https://www.rfc-editor.org/rfc/rfc9113#section-8.3.1

This behavioral change can be reverted by setting runtime flag envoy.reloadable_features.http2_discard_host_header to false.

Risk Level: Low
Testing: Unit Tests
Docs Changes: N/A
Release Notes: Yes
Platform Specific Features: N/A
Runtime guard: envoy.reloadable_features.http2_discard_host_header
Back-ports #30005, supporting #31118

### CODE REVIEW

File: changelogs/current.yaml [hunk 1]
```diff
@@ -2,6 +2,35 @@ date: Pending
...
+- area: stats dns_filter
+  change: |
+    Fixed tag extraction so that :ref:`stat_prefix <envoy_v3_api_msg_extensions.filters.udp.dns_filter.v3.DnsFilterConfig>`
+    is properly extracted. This changes the Prometheus name from
+    dns_filter_myprefix_local_a_record_answers{} to dns_filter_local_a_record_answers{envoy.dns_filter_prefix="myprefix"}.
+- area: http2
+  change: |
+    Changes the default value of ``envoy.reloadable_features.http2_use_oghttp2`` to true. This changes the codec used for HTTP/2
+    requests and responses. This behavior can be reverted by setting the feature to false.
+- area: http2
```
Comment (SeanKilleen) [inline changelogs/current.yaml:29]: ⚠️ Self-review: This is the only release note associated with the backport.  I'm unsure if the other notes should also be included here.

**Comment by phlax** [comment]
could you fix DCO please @SeanKilleen

File: changelogs/current.yaml (hunk 1 above)
Comment (phlax) [reply changelogs/current.yaml:29]: i would copy the file somewhere so you can easily grab the text, and then checkout the original from the branch - ie 

```console
$ git checkout release/v1.28 -- changelogs/current.yaml
```

with backports its ok to force push and preferably just a single commit - so best squash the fix into the original

**Comment by SeanKilleen** [comment]
I'm going to do the easiest thing in this case, close this PR, and attempt it cleanly in another PR.

**Pull request closed without merging**